XiEditor/EditView.swift

```diff
@@ -193,6 +193,22 @@ class EditView: NSView, NSTextInputClient {
             sendRpcAsync("request_lines", params: [f, l])
         }
 
+        // first pass, for drawing background selections
+        for lineIx in first..<last {
+            guard let line = getLine(lineIx), line.containsSelection == true else { continue }
+            let selections = line.styles.filter { $0.style == 0 }
+            let attrString = NSMutableAttributedString(string: line.text, attributes: self.attributes)
+            let ctline = CTLineCreateWithAttributedString(attrString)
+            let y = linespace * CGFloat(lineIx + 1)
+            context.setFillColor(selcolor().cgColor)
+            for selection in selections {
+                let selStart = CTLineGetOffsetForStringIndex(ctline, utf8_offset_to_utf16(line.text, selection.range.location), nil)
```

Comment:
I'm pretty sure you don't want the utf8_offset_to_utf16 call here (or on the following line), as the selection will already be in utf-16 units. I get a crash when doing selections over non-ascii text.